### DISCUSSION TITLE
Does `createResource` rerender after fetch got resolved?

### DISCUSSION THREAD

**Discussion by ngdangtu-vn** [solidjs/solid#1698] [Q&A]
I try to fetch data with `createResource` and `fetch`. Since API didn't help much, I had to push my luck on old document. Then I found this: https://docs.solidjs.com/guides/how-to-guides/routing-in-solid/solid-router#:~:text=here's%20an%20example%20of%20what%20this%20might%20look%20like%20in%20a%20createresource%20primitive

I try that code but it doesn't work because the data return when the fetching doesn't finish. Do I understand it wrong or the doc is outdate? How to make `createResource` return the final result?

I suspect I should use `createEffect`, but I don't really understand how it works. How does the hook know when to fire to get the correct data from `createResource` when I didn't pass any id to it?

**Comment by ryansolid** [comment]
I'm sorry that example isn't set up to listen to route changes. It should be:
```js
const params = useParams();
  const [data] = createResource(() => params.id, fetchUser) // 👈 Pass the dynamic route parameter to the createResource primitive
```

That being said even without the source that tracks updates it should fire the first time. You shouldn't need to use `createEffect`. Maybe a simple reproduction or code snippet would be useful to show what you are are trying to do and what isn't working?

**Reply by ngdangtu-vn** [reply]
Thanks, here is my code:

```typescript
function SectAlbum()
{
	const [gallery] = createResource(async _ =>
	{
		const res = await fetch('get/gallery.json')
		return res.json()
	})

	if (gallery.error) return <p>Unable to load albums 😢</p>
	if (gallery.loading) return <p>loading...</p>

	console.log(gallery.latest)

	const albumLs = gallery().albums as AlbumInfo[]
	return (
		<Section class='album-section' title='Albums'>
			<div class="decor decor-top" aria-hidden />
			<div class="decor decor-bottom" aria-hidden />
			<For each={albumLs}>{(album, i) => <AlbumFolder data={album} ref={`album-${i()}`} />}</For>
		</Section>
	)
}
```

In my case it returns `<p>loading...</p>` for once and finish. Even after the `get/gallery.json` is loaded, it is still not update.

**Reply by NightOnFire** [reply]
Solid only runs your component function one time and sets up reactivity for the jsx returned. If you return early from your component when its loading it will not hook up any of the stuff below it. This functionality should be accomplished using components such as Suspense and Show.
